runtime/criusupport/criusupport.cpp

```diff
@@ -376,13 +391,19 @@ Java_org_eclipse_openj9_criu_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
 			goto wakeJavaThreads;
 		}
 
+		vmFuncs->acquireExclusiveVMAccess(currentThread);
+		vm->memoryManagerFunctions->j9mm_iterate_all_objects(vm, PORTLIB, 0, objectIteratorCallback, vm->checkpointState.hookRecords);
+
```

Comment:
after doing the heapwalk, run the hooks on records that have `isRestore` set to false, `runInternalJVMCheckpointHooks(..)`

Reply:
Run `runInternalJVMCriuHooks(FALSE)` at checkpoint.